src/main/java/io/makerplayground/ui/canvas/CanvasView.java

```diff
@@ -336,6 +338,93 @@ private void zoomDefaultHandler() {
         mainPane.setScale(1);
     }
 
+    void setTopElement(NodeElement element, double upperbound,double joint)
+    {
+        if (element instanceof Condition || element instanceof Scene || element instanceof Delay) {
+            element.setTop(upperbound-joint);
+        }
+    }
+
+
+    double setDiagram(NodeElement element, double upperbound,double maxRangeSetting,double joint)
+    {
+        double rangeJoint = 0;
+        double totalRange = 0;
+
+        List<Line> lineList = new ArrayList<Line>();
+        for (Line line : canvasViewModel.getProject().getUnmodifiableLine()) {
+            if (line.getSource().equals(element)) {
+                lineList.add(line);
+            }
+        }
+
+        if (lineList.size() != 0) {
+            int rangeUserSetting = 0;
+            for (Line line : lineList) {
+                line.getDestination().setLeft(line.getSource().getLeft()+220);
+                if (line.getDestination() instanceof Condition) {
+                    int effectDevice = canvasViewModel.project.getDeviceWithCondition().size();  // get size of list that ProjectDevice has effect with ConditionViewModel
+                    rangeUserSetting = ((Condition) line.getDestination()).getSetting().size()+((Condition)line.getDestination()).getVirtualDeviceSetting().size();
```

Comment:
Try using sceneView.getBoundsInParent() to get the exact height of node as it is more robust to changes in node layout in the future